### PR TITLE
ignore duplicated bConfigurationValue

### LINKED ISSUE #54: AddDescriptor crashes adding duplicate key
When trying to run `./usbipd wsl list` I get:

```
Unhandled exception. System.AggregateException: One or more errors occurred. (An item with the same key has already been added. Key: [1, UsbIpServer.UsbConfigurationDescriptors+UsbConfiguration])
 ---> System.ArgumentException: An item with the same key has already been added. Key: [1, UsbIpServer.UsbConfigurationDescriptors+UsbConfiguration]
   at System.Collections.Generic.TreeSet`1.AddIfNotPresent(T item)
   at System.Collections.Generic.SortedSet`1.Add(T item)
   at System.Collections.Generic.SortedDictionary`2.Add(TKey key, TValue value)
   at UsbIpServer.UsbConfigurationDescriptors.AddDescriptor(ReadOnlySpan`1 descriptor) in C:\github\usbipd-win\UsbIpServer\UsbConfigurationDescriptors.cs:line 98
   at UsbIpServer.ExportedDevice.GetConfigurationDescriptor(DeviceFile hub, UInt16 connectionIndex, Byte numConfigurations) in C:\github\usbipd-win\UsbIpServer\ExportedDevice.cs:line 130
   at UsbIpServer.ExportedDevice.GetDevice(SafeDeviceInfoSetHandle deviceInfoSet, SP_DEVINFO_DATA devInfoData, CancellationToken cancellationToken) in C:\github\usbipd-win\UsbIpServer\ExportedDevice.cs:line 202
   at UsbIpServer.ExportedDevice.GetAll(CancellationToken cancellationToken) in C:\github\usbipd-win\UsbIpServer\ExportedDevice.cs:line 239
   at UsbIpServer.Program.<>c.<<Main>b__19_18>d.MoveNext() in C:\github\usbipd-win\UsbIpServer\Program.cs:line 387
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.<>c__DisplayClass79_0.<OnExecute>b__0()
   at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at UsbIpServer.Program.Main(String[] args) in C:\github\usbipd-win\UsbIpServer\Program.cs:line 555
```

So I added some Console.WriteLine logging and get

```
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Adding Key: 1 with Windows.Win32.Devices.Usb.USB_CONFIGURATION_DESCRIPTOR
Unhandled exception. System.AggregateException: One or more errors occurred. (An item with the same key has already been added. Key: [1, UsbIpServer.UsbConfigurationDescriptors+UsbConfiguration])
```

But I'm confused. Why didn't the second "adding key: 1" fail, and what's config.bConfigurationValue expected to be?

It seems like my machine has some number of devices that you're not expecting to "be the same" when loaded into the hashtable. Two mice? Two webcams? Two identical brand Anker USB Hubs? Yes to all of these things.

Any ideas on the best way to debug?

### PR DESCRIPTION
**Pull request by cho45**
Fix #54 

I am not sure if this is an accurate change. However, I believe it is better than the current situation where it crashes.

### CODE REVIEW

**Review by dorssel — CHANGES_REQUESTED** [review]
For now, this will quickly avoid most problems, but I expect that if you actually want to attach *this* device (i.e. the one with multiple configuration), then we do need to parse it properly instead of just ignoring it.

Unfortunately, I don't have any such devices. Are you able/willing to do some testing in the future once I figured this out?

File: UsbIpServer/UsbConfigurationDescriptors.cs [hunk 1]
```diff
@@ -94,7 +94,13 @@ public void AddDescriptor(ReadOnlySpan<byte> descriptor)
...
                         BytesToStruct(descriptor[offset..], out USB_CONFIGURATION_DESCRIPTOR config);
                         configuration = new(config);
-                        Configurations.Add(config.bConfigurationValue, configuration);
+                        if (Configurations.ContainsKey(config.bConfigurationValue))
+                        {
+                            Console.Error.WriteLine($"WARNING: bConfigurationValue {config.bConfigurationValue} is duplicated. ignored.");
+                        } else
+                        {
+                            Configurations.Add(config.bConfigurationValue, configuration);
+                        }
```
[the remark this answers is not in the document]
Comment (cho45) [reply UsbIpServer/UsbConfigurationDescriptors.cs]: Oh, you have the right method. I've almost never written C# before, so I appreciate the pointers.

**Comment by cho45** [comment]
Of course, it is good to be able to parse just fine in such a case.

I can test it and see if there are any errors. It may be difficult to connect to the actual device having the problem with usbip to try.

File: UsbIpServer/UsbConfigurationDescriptors.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dorssel) [reply UsbIpServer/UsbConfigurationDescriptors.cs]: Thx for the fix. It's ok to commit the suggestion yourself, but next time you may just want to "accept". Saves you the trouble of copy/paste.

**Review by dorssel — APPROVED** [review]

**Comment by shanselman** [comment]
I did some testing and can confirm that you can't attach to ANY device using usbipd attach if the dual config device is attached. It will hang here. The only solution is to remove the device (I turned mine off)

image

That said, you can manually bind, etc, etc, but for the very convenient "attach" command this still hangs in enumeration, perhaps on the linux side?

**Comment by dorssel** [comment]
@shanselman 
Are you willing to conduct an experiment?

This code (enumeration) is used not only in CLI, but also in the service. My guess is that the blunt `Console.Error.WriteLine` is interfering with the remote command execute. What if you just comment out the WARNING? I.e. just silently ignore?

I'm still looking for a way to reproduce this myself, but no luck yet.

**Comment by shanselman** [comment]
Lovely @dorssel, I see the latest release is 15 days ago, I'll install this one as soon as it's packaged up! Craig and I are going to do a promotional video on usbipd this week for my youtube!

**Comment by dorssel** [comment]
Cool. I was planning for a release tomorrow or Wednesday. We're polishing some of the more common issues, some just documentation, some real. By now, the latest reported issues are >50% known and fixed, so it is time to release those fixes alright.

**Comment by shanselman** [comment]
Lovely! We are recording at noon on Weds. Looking forward to the release!

**Pull request merged**